### DISCUSSION TITLE
ESP32 S3 mini (Zero)

### DISCUSSION THREAD

**Discussion by MichaelGer1410** [Trunks1982/Daikin-Rotex-HPSU-CAN#39] [Feedback] [answered]
Hi, habe ein kleines Problem mit der Rorex.bin.

Ich kann die Datei zwar Flashen, jedoch wird mir kein rotex wlan angezeigt.
Wenn ich allerdings die ESPHOME Firmware (Prepair for first USE) sehe ich das esphome-web.

Woran kann das liegen?

**Comment by Trunks1982** [comment]
Die bin Datei ist nur zum updaten.Du musst erst die Factory-rotex.bin flashen.

**Comment by MichaelGer1410** [comment]
Sorry, meinte doch die rotex factory. Die Rotex.bin ist ja nur für das OTA Update.
Ich bin schon seit 2 Stunden hier am Flaschen aber es will einfach nicht

**Reply by Trunks1982** [reply]
Also du flasht die und es kommt nach ca 2-3min kein Wlan Hotspot?

**Reply by MichaelGer1410** [reply]
Ja genau. das ESPHOME Web tool sagt mir erfolgreich geflasht aber kein Hotspot mit der Firmware von dir, nur wenn ich die Original von ESPHOME Flashe.

**Reply by Trunks1982** [reply]
Mhm

**Reply by Trunks1982** [reply]
Was benutzt du als Visualisierung?

**Reply by MichaelGer1410** [reply]
ioBroker-VIS

**Reply by Trunks1982** [reply]
Ok.Ich kann gerne gleich nochmal eine neue bin hochladen.Vielleicht stimmt da was nicht mit.Nirmalerweise sollte das funktionieren.

**Reply by MichaelGer1410** [reply]
Das wäre nett

**Comment by MichaelGer1410** [comment]
Ich hatte bisher immer Wemos D1 Mini, damit ging das alles etwas einfacher aus meiner sicht. Mit dem Ding habe ich noch keine erfahrung. Auch komisch das man diesen Boot knopf immer drücken muss...

**Reply by Trunks1982** [reply]
So hab jetzt nochmal die Datei neu hochgeladen.Also ich musste bei meine S3 noch nie nen Bootknopf drücken um zu flashen.
probier noch mal.
D1 Mini hab ich noch nie gehabt, aber diese Variante ist eigentlich total easy. Idioten sicher:-)

**Reply by MichaelGer1410** [reply]
Ja das dachte ich mir auch. Mal eben 2 teile bestellen, flashen und schon Läuft das mit der Rotex, ja sch**** :))

**Reply by MichaelGer1410** [reply]
So, ist geflasht, leider kein WLAN zu sehen. Muss ich die Rotex.bin danach auch noch flashen? ne oder?

**Reply by Trunks1982** [reply]
So ist es auch.Ich benutze aber keinen zero sondern einen ganz normalen Esp32 S3.Was mir gerade einfällt du hast glaub nicht den gpio pin den ich benutze.kann sein das es daran liegt.

**Reply by Trunks1982** [reply]
nein musst die rotex.bin dann nicht nochmal flashen.

**Reply by Trunks1982** [reply]
Welche pins hast du?
Wenn du den Zero hast hast du nicht die gleichen wie ich.

**Reply by MichaelGer1410** [reply]
> So ist es auch.Ich benutze aber keinen zero sondern einen ganz normalen Esp32 S3.Was mir gerade einfällt du hast glaub nicht den gpio pin den ich benutze.kann sein das es daran liegt.

Ich wollte es möglichst klein, daher habe ich den Zero genommen. Du hast ja geschrieben das der auch funktioniert. Angelötet habe ich an dem Ding noch nix, ist noch ganz nackig, nur USB Kabel ist dran.

**Reply by Trunks1982** [reply]
Funktioniert auch.Ich benutze den aber nicht und habe es auch nie getestet.Ein User benutzt den.Welche zwei pins ausser ground willst nutzen?

**Reply by MichaelGer1410** [reply]
![IMG_20240527_203929728](https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/assets/170225415/5a416faf-ff32-4c76-b36a-c7dd69fdc82f)
Such dir welche aus ;)

**Reply by Trunks1982** [reply]
So versuch mal die neue s3-zero und drück mal nichts beim flashen.

**Reply by MichaelGer1410** [reply]
So ist geflasht. Jetzt warte ich mal :D

**Reply by MichaelGer1410** [reply]
Tut sich auch nichts. Echt komisch. Oder hat das was mit dem USB Treiber zu tun den ich verwende? Anzeigen tut der bei mir YTAG

**Reply by Trunks1982** [reply]
Sollte usb serial stehn.

**Reply by MichaelGer1410** [reply]
ups steht nicht YTAG sondern "USB JTAG/Serial debug unit"

**Reply by MichaelGer1410** [reply]
https://www.amazon.de/ESP32-DevKitC-WROOM1-Entwicklungsboard-Bluetooth/dp/B0CLD4QKT1/ref=sr_1_3?dib=eyJ2IjoiMSJ9.KjUh5rNKVX4FXrbsymkzuMYYYxlAu007b1n1GwmD7i65fjhbPBsHZEh9izJ6qAjU0nCYOcJepqTltnO0xRm6DfYDOoYWGNtkB7xBbKNrYXvrGcr_HmJyGrjwXwmAmJwpJF6SbzGLWzjt2RizpXIckLy0KtqWBm-kAu3SB9cW3QZMhJsIuugyieUxhqqWOFXX5KLnKbyZMW-351ddrnNf9-uIKJ9Jn_VJkkKFl7CqTcw.sOURfS0iXWgLo24XaZJEyfJu_hokq32sleBmTVzihTg&dib_tag=se&keywords=esp32+s3&qid=1716836937&sr=8-3

ist das der den du hast? Dann bestelle ich mir den einfach mal und teste es damit :)

**Reply by Trunks1982** [reply]
Mhm.Dann weiss ich auch nicht weiter.Die Dateien funktionieren aufjedenfall.

**Reply by Trunks1982** [reply]
> https://www.amazon.de/ESP32-DevKitC-WROOM1-Entwicklungsboard-Bluetooth/dp/B0CLD4QKT1/ref=sr_1_3?dib=eyJ2IjoiMSJ9.KjUh5rNKVX4FXrbsymkzuMYYYxlAu007b1n1GwmD7i65fjhbPBsHZEh9izJ6qAjU0nCYOcJepqTltnO0xRm6DfYDOoYWGNtkB7xBbKNrYXvrGcr_HmJyGrjwXwmAmJwpJF6SbzGLWzjt2RizpXIckLy0KtqWBm-kAu3SB9cW3QZMhJsIuugyieUxhqqWOFXX5KLnKbyZMW-351ddrnNf9-uIKJ9Jn_VJkkKFl7CqTcw.sOURfS0iXWgLo24XaZJEyfJu_hokq32sleBmTVzihTg&dib_tag=se&keywords=esp32+s3&qid=1716836937&sr=8-3
> 
> ist das der den du hast? Dann bestelle ich mir den einfach mal und teste es damit :)

Nein hab den mit einer Cam.Der sollte aber auch gehn. ist ja auch mit s3 chip und die gpios sind auch da die ich benutze.

**Reply by MichaelGer1410** [reply]
> Mhm.Dann weiss ich auch nicht weiter.Die Dateien funktionieren aufjedenfall.

Aber danke erstmal für deine Hilfe! Bekommst morgen einen Kaffee spendiert!

**Reply by MichaelGer1410** [reply]
> > https://www.amazon.de/ESP32-DevKitC-WROOM1-Entwicklungsboard-Bluetooth/dp/B0CLD4QKT1/ref=sr_1_3?dib=eyJ2IjoiMSJ9.KjUh5rNKVX4FXrbsymkzuMYYYxlAu007b1n1GwmD7i65fjhbPBsHZEh9izJ6qAjU0nCYOcJepqTltnO0xRm6DfYDOoYWGNtkB7xBbKNrYXvrGcr_HmJyGrjwXwmAmJwpJF6SbzGLWzjt2RizpXIckLy0KtqWBm-kAu3SB9cW3QZMhJsIuugyieUxhqqWOFXX5KLnKbyZMW-351ddrnNf9-uIKJ9Jn_VJkkKFl7CqTcw.sOURfS0iXWgLo24XaZJEyfJu_hokq32sleBmTVzihTg&dib_tag=se&keywords=esp32+s3&qid=1716836937&sr=8-3
> > ist das der den du hast? Dann bestelle ich mir den einfach mal und teste es damit :)
> 
> Nein hab den mit einer Cam.Der sollte aber auch gehn. ist ja auch mit s3 chip und die gpios sind auch da die ich benutze.

CAM?
:EDIT:Aso Camera :))

**Reply by Trunks1982** [reply]
Vielleicht meldet sich gleich noch der User der den zero benutzt wie du

**Reply by MichaelGer1410** [reply]
> Vielleicht meldet sich gleich noch der User der den zero benutzt wie du
Das wäre super, vielleicht weis der was ich hier falsch mache...
So haue mich erstmal aufs Ohr

**Reply by Trunks1982** [reply]
Ja den esp32 s3 gabs mal mit cam zusatz.Hat aber nichts zusagen.Wollte damals nicht löten und der war schon fertig.

**Comment by ldnrktml** [comment]
Hi, hab den auch in Verwendung, anbei mal ein Bild mit ausgesteckter CAN Platine (hier noch nicht verlötet)
GND musste ich eine Leitung ziehen, die andren haben so gepasst.

![espzerso](https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/assets/99878337/27d1e710-9f2e-4cfc-8dce-b480659e271e). 

Die rotex.bin kann man nicht nehmen, da die GPIOs hier anders sind.

Ich habe das Teil im Home Assistant mit ESPHome geflasht und stets die GPIOs vor dem flashen in der yaml angepasst.

tx_pin: GPIO02
rx_pin: GPIO03

Allerdings sollte es auch ohne korrekte GPIOs funktionieren, zumindest der lokale Hotspot sollte erreichbar sein.

Den Boot Knopf habe ich nie genutzt, wann musst du den drücken?

**Reply by ldnrktml** [reply]
Hier mal meine bin, aus der aktuell laufenden yaml erstellt:

Bitte beachten, das CAN Interface muss in dem Fall wie auf dem Bild zu sehen verbunden sein, da:

tx_pin: GPIO02
rx_pin: GPIO03

entspricht.

[rotex-factory.zip](https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/files/15461309/rotex-factory.zip)

**Reply by MichaelGer1410** [reply]
Guten Morgen!
Deine bin habe ich auch mal getestet, geflasht wird, aber kein hotspot.
Habe auch mal mit dem ESPhome Flasher versucht: folgendes war der Output nach dem Flashen:
`Using 'COM3' as serial port.
Writing at 0x000f9e4a... (100 %)Wrote 960000 bytes (591110 compressed) at 0x00010000 in 9.1 seconds (effective 846.5 kbit/s)...
Hash of data verified.

Leaving...
Hard Resetting...
Hard resetting via RTS pin...
Done! Flashing is complete!

Showing logs:`

Und beim Booten: 
`[08:28:08]invalid header: 0xffffffff
[08:28:08]invalid header: 0xffffffff
[08:28:08]invalid header: 0xffESP-ROM:esp32s3-20210327
[08:28:08]Build:Mar 27 2021
[08:28:08]rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
[08:28:08]Saved PC:0x40048839
[08:28:08]invalid header: 0xffffffff
[08:28:08]invalid header: 0xffffffff
[08:28:08]invalid header: 0xffffffff
[08:28:08]invalid header: 0xffffffff
[08:28:08]invalid header: 0xffffffff
[08:28:08]invalid header: 0xffffffff`

Also wenn ich im Windows bin und nicht den Boot knopf drücke ist im sekundentakt das USB erkannt und wieder weg. Bis ich den Boot Knopf drücke. Dann kann ich Flashen.

**Reply by Trunks1982** [reply]
Dann kann ich dir da nicht weiterhelfen.Wie oben beschrieben hats der Iser gemacht und es hat geklappt.

**Reply by MichaelGer1410** [reply]
Soll morgen den ESP32-S3 bekommen. Versuche es dann damit nochmal. Melde mich dann mal. Danke!

**Reply by Trunks1982** [reply]
Ok👍

**Reply by MichaelGer1410** [reply]
Es wurde mir angezeigt das das Update erfolgreich war. 
Dann hab ich 5 min gewartet, aber ESP nicht erreichbar. Dann habe ich Ben Stromreset gemacht, ESP immer noch nicht erreichbar 😭

**Reply by Trunks1982** [reply]
Also wenn du Standard mässig alles gelassen hast funktioniert das.Nutze ich ja auch des öfteren.

**Reply by MichaelGer1410** [reply]
Ich kann nur sagen wie es ist. War bis vor dem Update noch an VIS mit dem ESP zugange. Jetzt geht nix mehr.
Wie gesagt er wird auch nicht mehr von PC via USB erkannt. Habe auch schon mehrere USB kabel versucht.

**Reply by Trunks1982** [reply]
Das er mit dem Usb Kabel nicht mehr gefunden wird hat nichts mit dem Update zu tun.Sobald Esp am Rechner angeschlossen wird funzt immer flashen.

**Reply by MichaelGer1410** [reply]
Wieso glaubstb du mir das nicht?

**Reply by Trunks1982** [reply]
Doch aber ich kann dir auch kur sagen das es nichts mit dem Update zu tun hat.Flashe meinen auch ziemlich oft hintereinander und der läuft immer egal was ich mache.Und wenn die bin nicht funktioniert hat, konnte ich ihn immer wieder am PC neuflashen. Was du nicht machen darfst im Browser die gleiche seite öfters öffnen.Dann wird der esp nicht gefunden.Es darf nur eine Seite geöffnet sein.Würde den Browser mal komplett schliessen.

**Reply by MichaelGer1410** [reply]
Flash die Rotex.bin mal selber über OTA.
Habe den ESP jetzt nochmal am 2. PC angeschlossen. Es wird nichts erkannt! PC macht keine anstalten das etwas angeschlossen wurde, im gerätemanager ist auch nix zu sehen.
Glaube da ist ein fehler im Update.

**Reply by Trunks1982** [reply]
Getestet.Läuft ohne Probleme.Kann direkt 5s nach Flash Vorgang die Seite öffnen.

**Reply by MichaelGer1410** [reply]
Kann doch nicht wahr sein.
Als du die datei hochgeladen hast, da war am VIS zugange und er funktionierte ohne Probleme. Nach Update war er tot. Da konnte selbst ich nix falsch machen ;)
Naja hab den bei A***** reklamiert.
Neuer kommt morgen

**Reply by Trunks1982** [reply]
Sehr merkwürdig.Hab gerade extra die bin von github geladen und ota geflasht.
Hast zwei USB Anschlüsse am esp?

**Reply by MichaelGer1410** [reply]
Ja 2, beide getestet, tot :(

::EDIT:: 
Sehe gerade das ich die ganze Zeit in dem falschen Thread poste

**Reply by Trunks1982** [reply]
🙈Gehn auch keine LED‘s an?

**Reply by MichaelGer1410** [reply]
Doch eine, rot. Wenn ich reset drücke bleibt die rot und die andere blizt nur ganz kurz auf.

**Reply by Trunks1982** [reply]
Can Chip ist angeschlossen am ESP?

**Reply by MichaelGer1410** [reply]
Hatte ich auch schon abgeklemmt um den auszuschliessen

**Reply by Trunks1982** [reply]
Ok.Sehr merkwürdig.

**Comment by MichaelGer1410** [comment]
Habe eben den neuen ESP32-S3 (nicht Mini) erhalten. Geflasht, und siehe da Rotex Hotspot vorhanden. Heute schaffe ich es leider nicht mehr weiter zu testen. Morgen werden ich den Rest mal in angriff nehmen.

**Reply by Trunks1982** [reply]
Top.Dann hat wohl der andere Esp einen weg.

**Reply by MichaelGer1410** [reply]
das denke ich auch!

**Reply by MichaelGer1410** [reply]
Sind die spenden eigenlich bei dir auf PP angekommen?

**Reply by Trunks1982** [reply, marked as answer]
Ja ganz vielen lieben Dank!

**Reply by MichaelGer1410** [reply]
Ich habe zu Danken 👍
